src/__tests__/utils/test-utils.tsx

```diff
@@ -37,18 +37,22 @@ export function renderWithRouter(
   ui: React.ReactElement,
   {
     route = '/',
-    history = createMemoryHistory({ initialEntries: [route] }),
+    search = '',
+    loader = undefined as undefined | LoaderFunction,
   } = {},
   theme: ThemeConfig = null,
 ) {
+  // jsdom doesn't like that we assign to location or location.href directly,
+  // because this is a navigation. Instead we use `replaceState` that isn't a
+  // navigation and works also for us.
+  window.history.replaceState(null, '', route + search);
   return {
     ...render(
-      <Router location={history.location} navigator={history}>
-        {ui}
-      </Router>,
+      <RouterProvider
+        router={createBrowserRouter([{ path: route, element: ui, loader }])}
```

Comment:
Note: I decided to use a browser router for tests because then it's easy to use window.location to check for the current location.

I also want to use a browser router for the app too later, but didn't want to implement it in this patch just yet.